great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -1921,17 +1929,19 @@ def _sqlalchemy_map_condition_unexpected_count_value(
         domain_kwargs=domain_kwargs,
     )
 
-    count_case_statement: List[sa.sql.elements.Label] = [
-        sa.case(
-            [
-                (
-                    unexpected_condition,
-                    sa.sql.expression.cast(1, sa.Numeric),
-                )
-            ],
-            else_=0,
-        ).label("condition")
-    ]
+    count_case_statement: List[Label] = sa.case(
+        [
+            (
+                unexpected_condition,
+                1,
```

Comment:
I happen to know that this was the way it was previously (`sa.sql.expression.cast(1, sa.Numeric)`) to avoid a Redshift bug in Issue [3293](https://github.com/great-expectations/great_expectations/issues/3293). I would recommend reverting to the way it was and adding a comment to avoid regression.

Reply:
@NathanFarmer Good catch.  Do we need the same `cast` for the `else_` condition?  Thanks!

Reply:
@alexsherstinsky I suspect yes. I believe I missed adding a Redshift test for this, so that would answer this question definitively.

Reply:
All done -- thanks!